core/primitives/src/congestion_info.rs

```diff
@@ -85,7 +85,7 @@ impl CongestionControl {
         // `clamped_f64_fraction` clamps to exactly 1.0.
         if congestion == 1.0 {
             // Red traffic light: reduce to minimum speed
-            if sender_shard == self.info.allowed_shard() as u64 {
+            if sender_shard.get() == self.info.allowed_shard() as u64 {
```

Comment:
for these kinds of operations (to hide the internal u64 more), would it make sense to not have `get()` but overriding the operation involving u64 (in this particular case for example overriding `==` op or adding `equals(u64)`)?

similarly for ops like `self.set_allowed_shard(allowed_shard.get() as u16);`, defining `into()` operations?